samples/sensor/tmp108/src/main.c

```diff
@@ -0,0 +1,181 @@
+ */
+#define ENABLE_TEMP_ALERT           0
+#define TEMP_ALERT_HIGH_VAL_IN_C    26
+#define TEMP_ALERT_LOW_VAL_IN_C     18
+
+void temperature_one_shot(const struct device *dev,
+			  struct sensor_trigger *trigger)
+{
+
+	struct sensor_value temp_value;
+	int result;
+
+	result = sensor_channel_get(dev,
+				    SENSOR_CHAN_AMBIENT_TEMP,
+				    &temp_value);
+
+	if (result) {
+		printf("error: sensor_channel_get failed: %d\n", result);
+		return;
+	}
+
+	printf("One shot power saving mode enabled, temperature is %gC\n", sensor_value_to_double(&temp_value));
+}
+
+void temperature_alert(const struct device *dev,
+		       struct sensor_trigger *trigger)
+{
+
+	struct sensor_value temp_flags = { 0 };
+
+	sensor_attr_get(dev,
+			SENSOR_CHAN_AMBIENT_TEMP,
+			SENSOR_ATTR_CONFIGURATION,
+			&temp_flags);
+
+	if (OVER_TEMP_FLAG & temp_flags.val1) {
+		printf("High temperature alert!\n");
+	} else if (UNDER_TEMP_FLAG & temp_flags.val1) {
+		printf("Low temperature alert!\n");
+	} else {
+		printf("Temperature alert cleared!\n");
+	}
+}
+
+void enable_temp_alerts(const struct device *tmp108)
+{
+
+	struct sensor_trigger sensor_trigger_type_temp_alert = {
+		.chan = SENSOR_CHAN_AMBIENT_TEMP,
+		.type = SENSOR_TRIG_THRESHOLD
+	};
+
+	struct sensor_value alert_upper_thresh = { TEMP_ALERT_HIGH_VAL_IN_C, 0 };
+	struct sensor_value alert_lower_thresh = { TEMP_ALERT_LOW_VAL_IN_C, 0 };
+	struct sensor_value alert_hysterisis = { 1, 0 };
+	struct sensor_value thermostat_mode = { 0, 0 };
+
+	sensor_attr_set(tmp108,
+			SENSOR_CHAN_AMBIENT_TEMP,
+			SENSOR_ATTR_ALERT,
+			&thermostat_mode);
+
+	sensor_attr_set(tmp108,
+			SENSOR_CHAN_AMBIENT_TEMP,
+			SENSOR_ATTR_HYSTERESIS,
+			&alert_hysterisis);
+
+	sensor_attr_set(tmp108,
+			SENSOR_CHAN_AMBIENT_TEMP,
+			SENSOR_ATTR_UPPER_THRESH,
+			&alert_upper_thresh);
+
+	sensor_attr_set(tmp108,
+			SENSOR_CHAN_AMBIENT_TEMP,
+			SENSOR_ATTR_LOWER_THRESH,
+			&alert_lower_thresh);
+
+	sensor_trigger_set(tmp108,
+			   &sensor_trigger_type_temp_alert,
+			   temperature_alert);
+}
+
+void enable_one_shot(const struct device *tmp108)
+{
+
+	struct sensor_trigger sensor_trigger_type_temp_one_shot = {
+		.chan = SENSOR_CHAN_AMBIENT_TEMP,
+		.type = SENSOR_TRIG_DATA_READY
+	};
+
+	sensor_attr_set(tmp108,
+			SENSOR_CHAN_AMBIENT_TEMP,
+			SENSOR_ATTR_ONE_SHOT_MODE,
+			NULL);
+
+	sensor_trigger_set(tmp108,
+			   &sensor_trigger_type_temp_one_shot,
+			   temperature_one_shot);
+}
+
+void get_temperature_continous(const struct device *tmp108)
+{
+
+	struct sensor_value temp_value;
+	int result;
+
+	result = sensor_channel_get(tmp108,
+				    SENSOR_CHAN_AMBIENT_TEMP,
+				    &temp_value);
+
+	if (result) {
+		printf("error: sensor_channel_get failed: %d\n", result);
+	}
+
+	printf("temperature is %gC\n", sensor_value_to_double(&temp_value));
+}
+
+void main(void)
+{
+	const struct device *tmp108;
+	int result;
+
+	printf("TI TMP108 Example, %s\n", CONFIG_ARCH);
+
+	tmp108 = DEVICE_DT_GET_ANY(ti_tmp108);
+
+	if (!tmp108) {
```

Comment:
This is not enough; you need to check if the sensor device was initialized correctly using `device_is_ready` here too.